### PR TITLE
[Batch 2] Notebook PR Ports.

### PR DESCRIPTION
**Pull request by Zsailer**
Don't merge without reading #96.

- [x] jupyter/notebook#3879
- [x] jupyter/notebook#3886
- [x] jupyter/notebook#3907
- [x] jupyter/notebook#4034
- [x] jupyter/notebook#4036

### CODE REVIEW

**Comment by kevin-bates** [comment]
No cleanup necessary.

File: jupyter_server/metrics.py [hunk 1]
```diff
@@ -1,18 +1,5 @@
-"""
-Prometheus metrics exported by Jupyter Notebook Server
+from ..prometheus.metrics import HTTP_REQUEST_DURATION_SECONDS
```
Comment (vidartf) [inline jupyter_server/metrics.py]: Should it really be a `..` here?

File: jupyter_server/prometheus/log_functions.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from notebook.prometheus.metrics import HTTP_REQUEST_DURATION_SECONDS
```
Comment (vidartf) [inline jupyter_server/prometheus/log_functions.py]: `notebook`

File: jupyter_server/services/kernels/kernelmanager.py [hunk 3]
```diff
@@ -26,6 +26,8 @@
 from jupyter_server._tz import utcnow, isoformat
 from ipython_genutils.py3compat import getcwd
 
+from notebook.prometheus.metrics import KERNEL_CURRENTLY_RUNNING_TOTAL
```
Comment (vidartf) [inline jupyter_server/services/kernels/kernelmanager.py]: notebook?

File: jupyter_server/prometheus/log_functions.py (hunk 2 above)
Comment (vidartf) [reply jupyter_server/prometheus/log_functions.py]: Fixed in https://github.com/jupyter/jupyter_server/pull/98.

File: jupyter_server/services/kernels/kernelmanager.py (hunk 3 above)
Comment (vidartf) [reply jupyter_server/services/kernels/kernelmanager.py]: https://github.com/jupyter/jupyter_server/pull/98

**Pull request merged**